solidity/contracts/bridge/Wallets.sol

```diff
@@ -299,4 +304,36 @@ library Wallets {
 
         emit NewWalletRegistered(ecdsaWalletID, walletPubKeyHash);
     }
+
+    /// @notice Reports about a proven fraud committed by the given wallet.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @dev Requirements:
+    ///      - Wallet must be in Live state
+    function notifyWalletFraudProven(
+        Data storage self,
+        bytes20 walletPubKeyHash
+    ) external {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+        require(
+            wallet.state == WalletState.Live ||
+                wallet.state == WalletState.MovingFunds,
+            "ECDSA wallet must be in Live or MovingFunds state"
+        );
+
+        wallet.state = WalletState.Terminated;
+
+        if (self.activeWalletPubKeyHash == walletPubKeyHash) {
+            // If termination refers to the current active wallet,
+            // unset the active wallet and make the wallet creation process
+            // possible in order to get a new healthy active wallet.
+            delete self.activeWalletPubKeyHash;
+        }
+
+        emit WalletTerminated(wallet.ecdsaWalletID, walletPubKeyHash);
+
+        // TODO: Perform slashing of wallet operators and add unit tests for that.
```

Comment:
Another argument for a separate function is that wallet which moved from `MovingFunds` to another state (unclear which and if we shouldn't add one more state) can be slashed if the transaction of moving funds is challenged. In this case, we will be slashing a wallet that is neither `Live` nor `MovingFunds`. At the same time, it makes only sense to terminate a wallet that is either `Live` or `MovingFunds` so the validation at line 318 makes sense.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/177#discussion_r833058231, this is why I planned to expose different public functions in `Wallets` lib to handle multiple wallet lifecycle events. That way, slashing, seizing, wallet termination, and friends can be totally hidden in the `Wallets` lib. The Bridge must only call the right function. Another benefit is reducing Bridge size since everything is kept inside `Wallets`.

Reply:
Ack, continuing the discussion in the other thread and resolving this one.